ethereum_transaction_data.go

```diff
@@ -20,121 +20,61 @@ package hedera
  *
  */
 
-import (
-	"encoding/json"
-
-	"github.com/pkg/errors"
-
-	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/rlp"
-)
-
 // Represents the data of an Ethereum transaction.
 type EthereumTransactionData struct {
-	eip1559 *types.DynamicFeeTx
-	legacy  *types.LegacyTx
+	eip1559 *EthereumEIP1559Transaction
+	legacy  *EthereumLegacyTransaction
 }
 
 // EthereumTransactionDataFromBytes constructs an EthereumTransactionData from a raw byte array.
 func EthereumTransactionDataFromBytes(b []byte) (*EthereumTransactionData, error) {
 	var transactionData EthereumTransactionData
-	if b[0] == 2 {
-		byt := b
-		byt = append(byt[:0], byt[0+1:]...)
-		err := rlp.DecodeBytes(byt, &transactionData.eip1559)
+	if b[0] == 0x02 {
+		eip1559, err := EthereumEIP1559TransactionFromBytes(b)
 		if err != nil {
 			return nil, err
 		}
 
+		transactionData.eip1559 = eip1559
 		return &transactionData, nil
 	}
 
-	err := rlp.DecodeBytes(b, &transactionData.legacy)
+	legacy, err := EthereumLegacyTransactionFromBytes(b)
```

Comment:
The legacy transaction flows are not tested. Could add similar tests for the legacy, as eip1559

Reply:
Will create a follow-up ticket for this.